src/ace.F90

```diff
@@ -760,123 +753,125 @@ subroutine read_reactions(nuc)
+            rxn % multiplicity_E % nbt = get_int(NR)
+            rxn % multiplicity_E % int = get_int(NR)
+          end if
 
-      ! read MT number, Q-value, and neutrons produced
-      rxn % MT            = int(XSS(LMT + i - 1))
-      rxn % Q_value       = XSS(JXS4 + i - 1)
-      rxn % multiplicity  = abs(nint(XSS(JXS5 + i - 1)))
-      rxn % scatter_in_cm = (nint(XSS(JXS5 + i - 1)) < 0)
-
-      ! Read energy-dependent multiplicities
-      if (rxn % multiplicity > 100) then
-        ! Set flag and allocate space for Tab1 to store yield
-        rxn % multiplicity_with_E = .true.
-        allocate(rxn % multiplicity_E)
-
-        XSS_index = JXS(11) + rxn % multiplicity - 101
-        NR = nint(XSS(XSS_index))
-        rxn % multiplicity_E % n_regions = NR
-
-        ! allocate space for ENDF interpolation parameters
-        if (NR > 0) then
-          allocate(rxn % multiplicity_E % nbt(NR))
-          allocate(rxn % multiplicity_E % int(NR))
-        end if
+          ! allocate space for yield data
+          XSS_index = XSS_index + 2*NR
+          NE = nint(XSS(XSS_index))
+          rxn % multiplicity_E % n_pairs = NE
+          allocate(rxn % multiplicity_E % x(NE))
+          allocate(rxn % multiplicity_E % y(NE))
 
-        ! read ENDF interpolation parameters
-        XSS_index = XSS_index + 1
-        if (NR > 0) then
-          rxn % multiplicity_E % nbt = get_int(NR)
-          rxn % multiplicity_E % int = get_int(NR)
+          ! read yield data
+          XSS_index = XSS_index + 1
+          rxn % multiplicity_E % x = get_real(NE)
+          rxn % multiplicity_E % y = get_real(NE)
         end if
 
-        ! allocate space for yield data
-        XSS_index = XSS_index + 2*NR
-        NE = nint(XSS(XSS_index))
-        rxn % multiplicity_E % n_pairs = NE
-        allocate(rxn % multiplicity_E % x(NE))
-        allocate(rxn % multiplicity_E % y(NE))
-
-        ! read yield data
-        XSS_index = XSS_index + 1
-        rxn % multiplicity_E % x = get_real(NE)
-        rxn % multiplicity_E % y = get_real(NE)
-      end if
-
-      ! read starting energy index
-      LOCA = int(XSS(LXS + i - 1))
-      IE   = int(XSS(JXS7 + LOCA - 1))
-      rxn % threshold = IE
-
-      ! read number of energies cross section values
-      NE = int(XSS(JXS7 + LOCA))
-      allocate(rxn % sigma(NE))
-      XSS_index = JXS7 + LOCA + 1
-      rxn % sigma = get_real(NE)
+        ! read starting energy index
+        LOCA = int(XSS(LXS + i - 1))
+        IE   = int(XSS(JXS7 + LOCA - 1))
+        rxn % threshold = IE
+
+        ! read number of energies cross section values
+        NE = int(XSS(JXS7 + LOCA))
+        allocate(rxn % sigma(NE))
+        XSS_index = JXS7 + LOCA + 1
+        rxn % sigma = get_real(NE)
+      end associate
     end do
 
     ! Create set of MT values
     do i = 1, size(nuc % reactions)
       call MTs % append(nuc % reactions(i) % MT)
+      call nuc%reaction_index%add_key(nuc%reactions(i)%MT, i)
```

Comment:
Perhaps you use whitespace when you are referencing a routine, but not an attribute?